ONSdigital/dp-api-tests
# Update e2e tests to encrypt/decrypt files on import process

### What

Updated e2e test to use aws sdk for go instead of goamz and use s3crypto package for decrypting or encrypting files sent sent/received from aws where necessary.

### How to review

Check test passes when `ENCRYPTION_DISABLED` is set to true and test passes when it is set to false; for the second test run you will need to export `RSA_PRIVATE_KEY` to both dp-dimension-extractor and dp-observation-extractor  and run those two services with`make acceptance`; dp-api-test will need the `RSA_PRIVATE_KEY`to be exported and `ENCRYPTION_DISABLED` set to be true

To create a private key do the following:
 1) Run `openssl genrsa -out private.pem<number_of_bytes>` where `number_of_bytes` can be 1024 or 2048
2) Add `export RSA_PRIVATE_KEY=$($HOME/<file_location>/<filename>.pem)` to your .bashrc or .bash_profile
3) Run `exec $SHELL -l` in terminal or open new terminal window to run `dp-dimension-extractor`, `dp-observation-extractor` and `dp-api-tests`
 
### Who can review

Team B
